cpp/src/io/parquet/predicate_pushdown.cpp

```diff
@@ -448,50 +448,24 @@ std::optional<std::vector<std::vector<size_type>>> aggregate_reader_metadata::fi
 
   // Converts AST to StatsAST with reference to min, max columns in above `stats_table`.
   stats_expression_converter stats_expr{filter.get(), static_cast<size_type>(output_dtypes.size())};
-  auto stats_ast     = stats_expr.get_stats_expr();
```

Comment:
Simply moved all this code block to `collect_filtered_row_group_indices()` function to reuse in `apply_bloom_filter()` as well